src/plots/cartesian/index.js

```diff
@@ -189,41 +195,60 @@ function plotOne(gd, plotinfo, cdSubplot, transitionOpts, makeOnCompleteCallback
     var modules = fullLayout._modules;
     var _module, cdModuleAndOthers, cdModule;
 
+    // Separate traces by zindex and plot each zindex group separately
+    // TODO: Performance
```

Comment:
Wondering if we could simplify this process by providing CSS z-index inside traces' plot step?

Reply:
I'm not following you here. To my understanding SVG doesn't have `z-index`, it always draws consecutive traces on top of the previous ones.

Reply:
That's right `z-index` is not supported in SVG.
Resolving.